pkgpanda documentation improvements.

orsenthil opened the pull request:
* Fixed typos.
* Fixed grammatical errors.
* Fixed formatting strings.
* Ensured that long lines break at `120` char and don't overflow the screen. (This is helpful as we improve the docs and `diff` will be legible).

I will iterate on this further as I gain more ground in understanding `pkgpanda`.

Please review @cmaloney and @spahl .

spahl commented:
:ship:

cmaloney commented:
Still need to review this, some of the docs should probably just be dropped entirely, and others have some chunks which I got the original wording really wrong as terms have shifted

lingmann commented on pkgpanda/docs/activating.md:
```diff
@@ -1,11 +1,13 @@
+# Activating Packages
 
-The list if currently active packages is stored in 'active.json' inside of INSTALL_ROOT.
+`pkgpanda activate <id>... [options]` will activate the list of packages give as `id`s in the command line.
```
I think something like `... will activate the list of package id's.` reads better.

lingmann commented on pkgpanda/docs/activating.md:
```diff
@@ -16,28 +18,34 @@ EnvironmentFile=/opt/mesosphere/config/mesos-slave-config
+    - Symlink config for each package into `INSTALL_ROOT/config/{provides}/`
+    - Aggregate environment variables to `INSTALL_ROOT/environment`
+
+5. Enable everything in `systemd`
```
Insert a blank line between 1 and sub-items or remove blank line between 2, 4 and sub-items to be consistent. Otherwise looks good and I find being explicit with the list numbers is generally less error-prone.

lingmann commented on pkgpanda/docs/architecture.md:
```diff
@@ -1,29 +1,64 @@
+**Config Generator**
 
-**Config Generator**: Given a list of enabled modules, and user-specified options (Resources, etc), generate a new config package.
+Given a list of enabled modules, and user-specified options (Resources, etc), generate a new config package.
```
This adjusted layout in `architecture.md` makes it a bit more difficult to identify subsections. For example, you can't really tell at a glance that each **Module** section is coupled with **Package Manager**, **Restart** etc. Try changing to bullet list with sub-sections?

lingmann commented on pkgpanda/docs/readme.md:
```diff
@@ -1,24 +1,28 @@
-[Modules](modueles.md)
+* [Architecture for Cluster Upgrades](architecture.md)
+* [Deployer component](deployer.md)
+* [Modules](modueles.md)
```
There is a typo in the original version of this... should be `modules.md`

lingmann commented on pkgpanda/docs/for_packagers.md:
```diff
@@ -1,109 +1,106 @@
+config applied to them. Pkgpanda in a future release will utilize [Systemd Sysusers] or something similar to provide
+the ability to manage users on a machine.
+
+[Systemd Sysusers]: (http://www.freedesktop.org/software/systemd/man/systemd-sysusers.html)
```
Remove `(` and `)` from link

lingmann commented on pkgpanda/docs/for_packagers.md:
```diff
@@ -1,109 +1,106 @@
+nothing which needs to be done, and the package can depend on the exact, predictable path of that file when unpacked as
+part of the package. It can specify the dependency on the full package-id.
 
 ## Depending on other packages
```
Remove (empty section)

lingmann commented on pkgpanda/docs/for_packagers.md:
```diff
@@ -1,109 +1,106 @@
+executables in path, files on filesystem, users). This is a much more robust way to state dependencies (At the cost of
+increased complexity, but the computation is still less than a second).
 
 ## Depending on other packages
```
Change to a top-level (`#`)

lingmann commented on pkgpanda/docs/for_packagers.md:
````diff
@@ -1,109 +1,106 @@
-Dependencies / requirements
+```bash
+tar -czf mesos--0.22.0.tar.xz -C mesos--0.22.0 .
+```
````
Remove `package command` section... does not seem useful.

lingmann commented on pkgpanda/docs/for_packagers.md:
```diff
@@ -1,109 +1,106 @@
+
+* environment variables
+* conflicts
+* Dependencies / requirements
```
Remove these above bullet points and add as a TODO line item:
` * Document package environment variables, conflicts, and dependencies / requirements`

lingmann commented:
Overall, this is a nice formatting improvement and and we should get it merged, even though there are parts of the documentation that are incorrect and need to be reworked. For example:

before:
![screen shot 2016-07-18 at 12 20 44 pm](https://cloud.githubusercontent.com/assets/33731/16927422/2104c7e2-4ce2-11e6-9f0f-54316cb4bffd.png)

after:
![screen shot 2016-07-18 at 12 21 29 pm](https://cloud.githubusercontent.com/assets/33731/16927435/31f4719c-4ce2-11e6-9f65-90a3d731f2a4.png)


I'm a :shipit: once above comments are addressed.

orsenthil commented:
I have addressed @lingmann  review comments.   Thanks for the review. 

I am still not familiar with pull request + merge process yet. Instead of all these merge commits, which got incurred when I updated my local branch, I am going to force-push my latest change on this branch, so that merging will be better. thanks.

lingmann commented:
Merged with #371
